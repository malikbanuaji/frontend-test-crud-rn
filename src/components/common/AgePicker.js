import {Picker} from '@react-native-community/picker';
import React, {useCallback, useEffect, useRef, useState} from 'react';

const MAX_AGE = 100;
const MIN_AGE = 1;

export default function AgePicker({style, selectedValue, onValueChange}) {
  const [age, setAge] = useState([]);

  const initList = useCallback(async () => {
    const _age = [{label: 'Umur', value: '0'}];
    for (let i = MIN_AGE; i <= MAX_AGE; i++) {
      _age.push({label: `${i}`, value: `${i}`});
    }
    setAge(_age);
  }, []);

  const ageRef = useRef();

  useEffect(() => {
    ageRef.current = initList;
  }, [initList]);

  useEffect(() => {
    ageRef.current();
  }, []);

  return (
    <Picker
      selectedValue={selectedValue}
      style={style}
      onValueChange={onValueChange}>
      {age.map((value) => (
        <Picker.Item key={value.value} {...value} />
      ))}
    </Picker>
  );
}
