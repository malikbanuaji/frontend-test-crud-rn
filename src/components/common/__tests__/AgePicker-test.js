import React from 'react';

import renderer from 'react-test-renderer';
import AgePicker from '../AgePicker';

test('renders correctly', () => {
  const tree = renderer.create(<AgePicker />).toJSON();
  expect(tree).toMatchSnapshot();
});
