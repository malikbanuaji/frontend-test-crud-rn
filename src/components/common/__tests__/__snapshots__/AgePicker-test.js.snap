// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <RNCPicker
    items={Array []}
    onChange={[Function]}
    selectedIndex={0}
    style={
      Array [
        Object {
          "height": 216,
        },
        undefined,
      ]
    }
  />
</View>
`;
